src/libsyncengine/jobs/network/abstractnetworkjob.cpp

```diff
@@ -573,26 +584,118 @@ bool AbstractNetworkJob::processSocketError(const std::string &msg, const Unique
+    const bool res = (stream.fail() || stream.bad()) && !stream.eof();
     if (res) {
         LOG_DEBUG(_logger, "sendRequest failed for job " << jobId() << " - stream fail=" << stream.fail()
                                                          << " - stream bad=" << stream.bad() << " - stream eof=" << stream.eof());
     }
     return res;
 }
 
-const std::string AbstractNetworkJob::errorText(const Poco::Exception &e) const {
+std::string AbstractNetworkJob::errorText(const Poco::Exception &e) const {
     std::ostringstream error;
     error << e.className() << " : (" << e.code() << ") : " << e.displayText().c_str();
     return error.str();
 }
 
-const std::string AbstractNetworkJob::errorText(const std::exception &e) const {
+std::string AbstractNetworkJob::errorText(const std::exception &e) const {
     std::ostringstream error;
     error << "(" << e.what() << ")";
     return error.str();
 }
 
+bool AbstractNetworkJob::handleJsonResponse(std::istream &is) {
+    // Extract JSON
+    try {
+        _jsonRes = Poco::JSON::Parser().parse(is).extract<Poco::JSON::Object::Ptr>();
+    } catch (Poco::Exception &exc) {
+        LOG_DEBUG(_logger,
+                  "Reply " << jobId() << " received doesn't contain a valid JSON payload: " << exc.displayText().c_str());
+        _exitCode = ExitCode::BackError;
+        _exitCause = ExitCause::ApiErr;
+        return false;
+    }
+
+    // Extract reply
+    std::ostringstream os;
+    _jsonRes->stringify(os);
+    if (isExtendedLog()) {
+        LOGW_DEBUG(_logger, L"Reply " << jobId() << L" received: " << Utility::s2ws(os.str()).c_str());
+    }
+    return true;
+}
+
+bool AbstractNetworkJob::handleJsonResponse(std::string &str) {
+    // Extract JSON
+    try {
+        Poco::JSON::Parser jsonParser;
+        _jsonRes = jsonParser.parse(str).extract<Poco::JSON::Object::Ptr>();
+    } catch (Poco::Exception &exc) {
+        LOG_DEBUG(_logger, "Reply " << jobId() << " received doesn't contain a valid JSON error: " << exc.displayText().c_str());
+
+        _exitCode = ExitCode::BackError;
+        _exitCause = ExitCause::ApiErr;
+        return false;
+    }
+
+    // Extract reply
+    if (isExtendedLog()) {
+        LOGW_DEBUG(_logger, L"Reply " << jobId() << L" received: " << Utility::s2ws(str).c_str());
+    }
+
+    return true;
+}
+
+bool AbstractNetworkJob::handleOctetStreamResponse(std::istream &is) {
+    getStringFromStream(is, _octetStreamRes);
+    return true;
+}
+
+bool AbstractNetworkJob::extractJsonError(std::istream &is, Poco::JSON::Object::Ptr errorObjPtr /*= nullptr*/) {
+    // Manage Content-Encoding
+    std::string encoding;
+    try {
+        encoding = _resHttp.get("Content-Encoding");
+    } catch (...) {
+        // No Content-Encoding
+    }
+
+    std::stringstream ss;
+    if (encoding == "gzip") {
+        unzip(is, ss);
+    } else {
+        ss << is.rdbuf();
+    }
+
+    Poco::JSON::Object::Ptr jsonObj;
```

Comment:
Most of the code below looks very similar to what we see in `AbstractNetworkJob::handleJsonResponse` implementation.

Reply:
Done